controllers/clustergroupupgrade_controller.go

```diff
@@ -1023,6 +1189,78 @@ func (r *ClusterGroupUpgradeReconciler) reconcileResources(ctx context.Context,
 	return nil
 }
 
+func (r *ClusterGroupUpgradeReconciler) getPolicyClusterStatus(policy *unstructured.Unstructured) ([]interface{}, error) {
+	policyName := policy.GetName()
+
+	// Get the compliant status part of the policy.
+	if policy.Object["status"] == nil {
+		return nil, fmt.Errorf("Policy %s is missing its status", policyName)
```

Comment:
A policy could be missing its status when it doesn't have a placement rule or up until the field is populated once the policy was matched with a placement rule. At the point this function is run, if a policy doesn't have its status then it means something pretty bad happened to the system, considering we had the 

I think it would be a good idea to treat some of these errors with temporary conditions not just here, but throughout the code with most of the ACM resources. I would rather put this on the list with other future improvements we want to do.